buf/registry/plugin/v1beta1/plugin.proto

```diff
@@ -0,0 +1,142 @@
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The time the Plugin was created on the BSR.
+  google.protobuf.Timestamp create_time = 2 [(buf.validate.field).required = true];
+  // The last time the Plugin was updated on the BSR.
+  google.protobuf.Timestamp update_time = 3 [(buf.validate.field).required = true];
+  // The name of the Plugin.
+  //
+  // Unique within a given Organization.
+  string name = 4 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 100
+  }];
+  // The id of the Organization that owns the Plugin.
+  string owner_id = 5 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The Plugin's visibility, either public or private.
+  PluginVisibility visibility = 6 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The Plugin state, either active or deprecated.
+  PluginState state = 7 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The Plugin type.
+  PluginType type = 8 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The configurable short description of the Plugin.
+  string description = 9 [(buf.validate.field).string.max_len = 350];
+  // The configurable source URL of the Plugin.
+  string source_url = 10 [
+    (buf.validate.field).string.uri = true,
+    (buf.validate.field).string.max_len = 255,
+    (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED
+  ];
+  // Managed indicates whether a Plugin is managed by the BSR.
+  //
+  // Managed Plugins are automatically updated and should not be updated by users because they will
+  // be overwritten.
+  bool managed = 11;
+  // The collections the Plugin belongs to.
+  repeated Collection collections = 12;
+}
+
+// The visibility of a Plugin, currently either public or private.
+enum PluginVisibility {
+  PLUGIN_VISIBILITY_UNSPECIFIED = 0;
+  // PLUGIN_VISIBILITY_PUBLIC says that the plugin is publicly available.
+  PLUGIN_VISIBILITY_PUBLIC = 1;
+  // PLUGIN_VISIBILITY_PRIVATE says that the plugin is private.
+  PLUGIN_VISIBILITY_PRIVATE = 2;
+}
+
+// The state of a Plugin, currently either active or deprecated.
+enum PluginState {
+  PLUGIN_STATE_UNSPECIFIED = 0;
+  // PLUGIN_STATE_ACTIVE says that the Plugin is currently active.
+  PLUGIN_STATE_ACTIVE = 1;
+  // PLUGIN_STATE_DEPRECATED says that the Plugin has been deprecated and should no longer be used.
+  PLUGIN_STATE_DEPRECATED = 2;
+}
+
+// The type of the Plugin.
+enum PluginType {
+  PLUGIN_TYPE_UNSPECIFIED = 0;
+  // PLUGIN_TYPE_PROTOC_CODE_GENERATION says that the Plugin is a protoc code generation plugin. It
+  // is intentionally a separate type because it does not use the pluginrpc protocol.
+  PLUGIN_TYPE_PROTOC_CODE_GENERATION = 1;
+  // PLUGIN_TYPE_CODE_GENERATION says that the Plugin is a code generation plugin.
+  PLUGIN_TYPE_CODE_GENERATION = 2;
```

Comment:
We don't have generation plugins at this time, why is this here? And why is `CodeGenerationConfig` here?

Reply:
Do not remove protoc plugins yet - the question about why or why not it is in there is not answered.

Reply:
> I thought the intention was to have a new upload service that would support both? Otherwise, we'll need to implement a stable buf command that relies on v1alpha1 CreateCuratedPlugin (docker) and v1 Upload (wasm),

Regardless of whether or not the new v1 API understands both legacy protoc and new check/generate plugins, the intention was definitely not for the new v1 API to support upload of both. Upload of legacy plugins should be left to the v1alpha1 API, or split out into a separate package. Many stable buf commands already rely on v1alpha1 APIs as it is, but regardless we should not dirty the new v1 `plugin` package with legacy plugin uploads based on docker.

Reply:
Okay great.

> Upload of legacy plugins should be left to the v1alpha1 API

@emcfarlane relevant when we get to `buf plugin push --image-{digest,config}`.